modules/apps/dispatch/dispatch-service/src/main/java/com/liferay/dispatch/internal/helper/DispatchTriggerHelper.java

```diff
@@ -37,9 +41,30 @@ public void addSchedulerJob(
 			String timeZoneId)
 		throws DispatchTriggerSchedulerException {
 
+		Date now = new Date();
+		Date startDate = dispatchTrigger.getStartDate();
+
+		if ((startDate == null) || startDate.before(now)) {
+			startDate = now;
+		}
+
+		if (dispatchTrigger.getCronExpression() != null) {
+			try {
+				CronExpression ce = new CronExpression(
+					dispatchTrigger.getCronExpression());
+
+				startDate = ce.getNextValidTimeAfter(startDate);
```

Comment:
I am not sure if importing the quartz library is a good idea, maybe we can avoid this by asking Core Infra to provide us a way to calculate this. Let me discuss it internally